### PR TITLE
[gRPC/Fix] Decoupling gRPC IDL dependencies @open sesame 12/17 10:05

### PR DESCRIPTION
**Pull request by dongju-chae**
This patch seperates gRPC IDL dependencies (protobuf/flatbuf) using dlopen().
From now on, gRPC plugin does not have mandatory dependencies for its IDLs.

It resolves https://github.com/nnstreamer/nnstreamer/issues/2931

Signed-off-by: Dongju Chae <dongju.chae@samsung.com>

### CODE REVIEW

**Review by myungjoo — APPROVED** [review]
LGTM

File: ext/nnstreamer/extra/nnstreamer_grpc_common.cc [hunk 1]
```diff
@@ -13,37 +13,66 @@
...
+    name = NNS_GRPC_PROTOBUF_NAME;
+  else if (idl == GRPC_IDL_FLATBUF)
+    name = NNS_GRPC_FLATBUF_NAME;
+
+  if (name == NULL) {
+    ml_loge ("Unsupported IDL detected: %d\n", idl);
+    return NULL;
   }
+
+  void * handle = dlopen (name, RTLD_LAZY);
```
Comment (jaeyun-jung) [inline ext/nnstreamer/extra/nnstreamer_grpc_common.cc:44]: how about using gmodule?

Comment (dongju-chae) [reply ext/nnstreamer/extra/nnstreamer_grpc_common.cc:44]: Oh, it looks better. thanks.

File: ext/nnstreamer/extra/nnstreamer_grpc_common.h [hunk 2]
```diff
@@ -23,6 +23,7 @@
 #include <cstring>
 #include <string>
 #include <thread>
+#include <dlfcn.h>
```
Comment (jaeyun-jung) [inline ext/nnstreamer/extra/nnstreamer_grpc_common.h:26]: is it necessary?

**Review by jaeyun-jung — APPROVED** [review]

File: ext/nnstreamer/extra/nnstreamer_grpc_common.h (hunk 2 above)
Comment (dongju-chae) [reply ext/nnstreamer/extra/nnstreamer_grpc_common.h:26]: Thanks, it's not necessary now.

**Pull request merged**